projectq/ops/_basics.py

```diff
@@ -295,23 +295,29 @@ def get_merged(self, other):
 
     def __eq__(self, other):
         """ Return True if same class and same rotation angle. """
-        tolerance = 1.e-12
-        if isinstance(other, self.__class__):
-            difference = abs(self._angle - other._angle)
```

Comment:
That will give the wrong final result when self._angle is 4pi-epsilon and other._angle is zero, The difference will end up being large instead of small, despite the angles being close.